### PR TITLE
Added zh-HANT localization

### PR DESCRIPTION
**Pull request by gwynjudd**
:warning:

Based on #255

### CODE REVIEW

**Comment by BenjaminVanRyseghem** [comment]
your commit looks to have too many things
(and to contain another PR)

**Comment by gwynjudd** [comment]
Sorry yes I wasn't quite sure what is the best thing to do when one change is requiring another fix in order to compile. This one needs the four character country code fix in PR 255. What do you prefer me to do?

On 27 Mar. 2017 22:05, Benjamin Van Ryseghem <notifications@github.com> wrote:

your commit looks to have too many things
(and to contain another PR)

**Comment by BenjaminVanRyseghem** [comment]
@gwynjudd I flagged (almost) all PRs requiring another one first.

I kinda like it better this way, as it avoids conflicts.

In all cases, if you want to ship both because it's needed, don't do it in a single commit (as git will have really hard time handling that :smile:)

File: languages/zh-HANT.js [hunk 1]
```diff
@@ -0,0 +1,37 @@
+/*!
+ * to be compatible with numbro.js language configuration
```
Comment (gwynjudd) [inline languages/zh-HANT.js]: Sorry about that it's just boilerplate from my colleague

**Comment by gwynjudd** [comment]
@BenjaminVanRyseghem should be resolved now

**Comment by BenjaminVanRyseghem** [comment]
Can you remove the commit "Allow four character country code" from this PR?

**Comment by BenjaminVanRyseghem** [comment]
shouldn't the code be

- zh-CN for Simplified Chinese
- zh-TW for Traditional Chinese

as said [here](https://github.com/foretagsplatsen/numbro/issues/238#issuecomment-289520442)?

**Comment by BenjaminVanRyseghem** [comment]
this looks pretty much like this [language](https://github.com/foretagsplatsen/numbro/blob/develop/languages/zh-TW.js) or?

**Comment by gwynjudd** [comment]
@BenjaminVanRyseghem yes it is very similar (the currency symbol is slightly different). I think it is meant to act as a fallback language for locales which use traditional chinese for which there isn't a specific fallback, e.g. Macau before the other patches. We are using zh-HANT and zh-HANS in our company for this purpose instead of zh-TW and zh-CN, so it is convenient for us to have them available as localizations.

**Comment by gwynjudd** [comment]
@BenjaminVanRyseghem "Allow four character country code" is removed

**Comment by gwynjudd** [comment]
@BenjaminVanRyseghem another way to do it may be to make them as aliases

```
window.numbro.culture('zh-HANT', language);
window.numbro.culture('zh-HANT-tw', language);
window.numbro.culture('zh-TW', language);
```

etc...

**Comment by ArmorDarks** [comment]
@gwynjudd @BenjaminVanRyseghem 

Ops, I missed this PR. See my explanations regarding `zh-TW` and `zh-Hant` here https://github.com/foretagsplatsen/numbro/pull/256#issuecomment-289907150.

So, in short, with name of locale in this PR there are few issues:

1. `zh-HANT` should be `zh-Hant`. `HANT` isn't country code, and caps for anything except country code violates standard.
2. `zh-Hant` isn't nor locale, nor culture, it is language code, and it is unpreferable to specify only language code for l10n files.
3. If we really want to keep `Hant` part, despite it is same as saying `zh-TW`, it should be `zh-Hant-TW`, not just `zh-TW`. However, if we want to do this, we should implement aliases or loose matching, as explained here https://github.com/foretagsplatsen/numbro/pull/256#issuecomment-289908620 and already suggested @gwynjudd

**Comment by gwynjudd** [comment]
@ArmorDarks @BenjaminVanRyseghem I think I've probably changed my mind on this one, I think we should revert it. As you say it's not really a proper locale, and I can accomplish my goal without requiring this fix. Sorry to be a bother

**Comment by BenjaminVanRyseghem** [comment]
ok, reverting :smile:

no problemo

**Comment by BenjaminVanRyseghem** [comment]
in numbro 1.11.0

**Pull request merged**